src/connector/sqlite.rs

```diff
@@ -50,25 +49,17 @@ impl TryFrom<&str> for SqliteParams {
         if path.is_dir() {
             Err(Error::builder(ErrorKind::DatabaseUrlIsInvalid(path.to_str().unwrap().to_string())).build())
         } else {
-            let official = vec![];
             let mut connection_limit = None;
             let mut db_name = None;
             let mut socket_timeout = None;
 
             if path_parts.len() > 1 {
-                let (_, unsupported): ConnectionParams = path_parts
-                    .last()
-                    .unwrap()
-                    .split('&')
-                    .map(|kv| {
-                        let splitted: Vec<&str> = kv.split('=').collect();
-                        (String::from(splitted[0]), String::from(splitted[1]))
-                    })
-                    .collect::<Vec<(String, String)>>()
-                    .into_iter()
-                    .partition(|(k, _)| official.contains(&k.as_str()));
-
-                for (k, v) in unsupported.into_iter() {
+                let params = path_parts.last().unwrap().split('&').map(|kv| {
+                    let splitted: Vec<&str> = kv.split('=').collect();
+                    (String::from(splitted[0]), String::from(splitted[1]))
```

Comment:
nitpick: do these need to be allocated, or would matching on string slices below be enough, and only allocating what we need?

Reply:
Done like donuts.